### DISCUSSION TITLE
What is TypeScript event type of ConfirmPopup click handler?

### DISCUSSION THREAD

**Discussion by liborjelinek** [primefaces/community#451] [PrimeVue] [answered]
In my TypeScript project, I am trying to follow the [basic ConfirmPopup example](https://primevue.org/confirmpopup/#basic).

```
<template>
  <Button icon="pi pi-trash" outlined v-tooltip.bottom="'Delete file'" @click="confirm1($event)" />
  <ConfirmPopup></ConfirmPopup>
</template>

<script setup lang="ts">
...

const confirm1 = (event) => {
  confirm.require({
    target: event.currentTarget,
    ....
```

However, the error is:

![image](https://github.com/primefaces/community/assets/383059/4bce9c47-4485-4efe-add9-e287dee6dd5c)

By placing a breakpoint in runtime, I figure out it's `PointerEvent`:

```
const confirm1 = (event: PointerEvent) => {
```

But now error on `target` is:

![image](https://github.com/primefaces/community/assets/383059/9780b4fa-a45d-4cad-9d13-435b35707e04)

How to correctly use ConfirmPopup click handler in TypeScript? Thanks 🙏

**Comment by FlipWarthog** [comment, marked as answer]
`EventTarget` implements `HTMLElement`; are you able to simply do `target: event.currentTarget as HTMLElement,`?

See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement

**Reply by liborjelinek** [reply]
No TS errors! Thank you!

![image](https://github.com/primefaces/community/assets/383059/5f57c6ce-4aa6-4792-bda0-b5948d72f35f)

**Reply by leonardocustodio** [reply]
The documentation definitely needs to be improved here: https://primevue.org/confirmpopup
Had to come here to find out how to use the component
